File: 724_Find Pivot Index/xsy/solution.py
class Solution:
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        lsum =  0
        rsum = sum(nums)
        for i in range(len(nums)):
            t = nums[i]
            rsum -= t
            if lsum == rsum:
                return i
            lsum += t
        return -1

File: 724_Find Pivot Index/xsy/solution.go
func sum(nums *[]int) int {
	s := 0
	for _, v := range *nums {
		s += v
	}
	return s
}


func pivotIndex(nums []int) int {
	lsum := 0
	rsum := sum(&nums)
	for i := 0; i < len(nums); i++ {
		t := nums[i]
		rsum = rsum - t
		if rsum == lsum {
			return i
		}
		lsum = lsum + t
	}
	return -1
}
